nexus_constructor/instrument_view_axes.py

```diff
@@ -0,0 +1,93 @@
+        """
+        Class for housing the objects that create the axes in the instrument view.
+        :param component_root_entity: The root entity for the instrument view components.
+        :param line_length: The length of the line in the axes.
+        """
+
+        self.component_root_entity = component_root_entity
+        self.line_length = line_length
+
+        # The coordinate array given to the LineGeometry must be in the form
+        # [start_x, start_y, start_z, end_x, end_y, end_z]. All lines start at the origin so the first three elements
+        # in the lists for the three lines will be the same.
+        origin = [0, 0, 0]
+        x_line_vertices = origin + [line_length, 0, 0]
+        y_line_vertices = origin + [0, line_length, 0]
+        z_line_vertices = origin + [0, 0, line_length]
+
+        self.x_line_geometry = LineGeometry(
+            QtCore.QByteArray(self.create_data_array(x_line_vertices))
+        )
+        self.y_line_geometry = LineGeometry(
+            QtCore.QByteArray(self.create_data_array(y_line_vertices))
+        )
+        self.z_line_geometry = LineGeometry(
+            QtCore.QByteArray(self.create_data_array(z_line_vertices))
+        )
+
+        self.x_line_mesh = Qt3DRender.QGeometryRenderer()
+        self.y_line_mesh = Qt3DRender.QGeometryRenderer()
+        self.z_line_mesh = Qt3DRender.QGeometryRenderer()
+
+        self.x_line_mesh.setPrimitiveType(Qt3DRender.QGeometryRenderer.Lines)
+        self.y_line_mesh.setPrimitiveType(Qt3DRender.QGeometryRenderer.Lines)
+        self.z_line_mesh.setPrimitiveType(Qt3DRender.QGeometryRenderer.Lines)
+
+        self.x_line_mesh.setGeometry(self.x_line_geometry)
+        self.y_line_mesh.setGeometry(self.y_line_geometry)
+        self.z_line_mesh.setGeometry(self.z_line_geometry)
+
+        self.x_line_entity = Qt3DCore.QEntity(self.component_root_entity)
+        self.y_line_entity = Qt3DCore.QEntity(self.component_root_entity)
+        self.z_line_entity = Qt3DCore.QEntity(self.component_root_entity)
+
+        self.x_line_material = Qt3DExtras.QPhongMaterial()
+        self.y_line_material = Qt3DExtras.QPhongMaterial()
+        self.z_line_material = Qt3DExtras.QPhongMaterial()
+
+        set_material_properties(
+            self.x_line_material, AxisColors.X.value, AxisColors.X.value
+        )
+        set_material_properties(
+            self.y_line_material, AxisColors.Y.value, AxisColors.Y.value
+        )
+        set_material_properties(
+            self.z_line_material, AxisColors.Z.value, AxisColors.Z.value
+        )
+
+        add_qcomponents_to_entity(
+            self.x_line_entity, [self.x_line_mesh, self.x_line_material]
+        )
+        add_qcomponents_to_entity(
+            self.y_line_entity, [self.y_line_mesh, self.y_line_material]
+        )
+        add_qcomponents_to_entity(
+            self.z_line_entity, [self.z_line_mesh, self.z_line_material]
+        )
```

Comment:
In general this whole thing is just doing the same thing 3 different times, we should be able to get it into a loop. Maybe something like:
```
self.geometries = dict()
self.line_meshes = dict()
self.materials = dict()

for i, colour in enumerate([AxisColors.X.value, AxisColors.Y.value, AxisColors.Z.value]):
    # Calculates the end x, y and z for each line in turn
    vertices = [0] * 3
    vertices[i] = line_length

    self.geometries[i] = LineGeometry(
        QtCore.QByteArray(self.create_data_array(vertices))
    )
    self.line_meshes[i] = Qt3DRender.QGeometryRenderer()
    self.line_meshes[i].setPrimitiveType(Qt3DRender.QGeometryRenderer.Lines)
    self.line_meshes[i].setGeometry(self.geometries[i])

    self.materials[i] = Qt3DExtras.QPhongMaterial()

    set_material_properties(
        self.materials[i], colour, colour
    )

    add_qcomponents_to_entity(
        Qt3DCore.QEntity(component_root_entity), [self.line_meshes[i], self.materials[i]]
    )
```
But that sacrifices some readability, what do you think?